common/djangoapps/util/views.py

```diff
@@ -201,28 +197,19 @@ def reset_course_deadlines(request):
     Set the start_date of a schedule to today, which in turn will adjust due dates for
     sequentials belonging to a self paced course
     """
-    from lms.urls import RENDER_XBLOCK_NAME
-    from openedx.features.course_experience.urls import COURSE_HOME_VIEW_NAME
+    referrer = request.META.get('HTTP_REFERER')
+    if not referrer:
+        return
 
-    detail_id_dict = ast.literal_eval(request.POST.get('reset_deadlines_redirect_url_id_dict'))
-    redirect_url = request.POST.get('reset_deadlines_redirect_url_base', COURSE_HOME_VIEW_NAME)
-    course_key = CourseKey.from_string(detail_id_dict['course_id'])
```

Comment:
This was a refactor to avoid having to pass the origin page, and simply use the referrer. Seemed easier.